platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -123,8 +125,9 @@ class HangingProtocolService {
     // this.protocols is a map of protocols with the protocol id as the key
     // and the protocol or a function that returns a protocol as the value
     const protocols = [];
+    const keys = this.activeProtocolIds || this.protocols.keys();
```

Comment:
this will always return `this.activeProtocolIds`, did you want to check `.length`?

Reply:
No, activeProtocolIds can be null, which means to use everything.